components/inputs/input-date.js

```diff
@@ -144,6 +146,15 @@ class InputDate extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(LitEl
 		this._dateTimeDescriptor = getDateTimeDescriptorShared();
 	}
 
+	disconnectedCallback() {
```

Comment:
@dbatiste I noticed that tabs also uses a ResizeObserver in firstUpdated and then doesn't use disconnectedCallback. Is this piece here necessary or can I remove it?

Reply:
So I don't _think_ so.  From a garbage collection perspective, the element and the handler being observed are all internal to the component, so when the component is GC'd I think the observer will be too.  

Aside from GC, functionally, `disconnectedCallback` doesn't necessarily mean that the component is going to be thrown away... i.e. the consumer may reattach it to the DOM.  In such case, there would need to be code in the `connectedCallback` to setup the observer again, assuming it was disconnected in the `disconnectedCallback`, except when `connectedCallback` is first run, the shadowDOM won't exist. The best an cleanest place to hook up listeners and observers is indeed `firstUpdated`, leaving them intact regardless of whether how many times the element is disconnected and reconnected.

Reply:
Thanks for the detailed explanation. It sounds like it's reasonably safe to remove the `disconnectedCallback` then.